src/%IPM/Main.cls

```diff
@@ -2383,11 +2412,229 @@ ClassMethod UpdateLanguageExtensions(pVerbose As %Boolean = 0, pTestOnly As %Boo
+			If currentNS = "" {
+				Quit
+			}
+			// avoid %SYS and implied namespaces
+			If (currentNS '= "%SYS") && ($ZCONVERT(currentNS, "U") = currentNS) {
+				New $Namespace
+				Set $Namespace = currentNS
+				If '$System.CLS.IsMthd("%IPM.Main", "Shell") {
+					Set namespacesForIPM(currentNS) = 1
+				}
+			}
+		}
+	}
+	Set $Namespace = baseNS
+	If globally {
+		Merge targetNamespaces = namespacesForIPM
+	} ElseIf (namespaces '= "") {
+		Set specifiedNSs = $ListFromString(namespaces, ",")
+		Set ptr = 0
+   		While $ListNext(specifiedNSs, ptr, value) {
+			If '$DATA(namespacesForIPM(value)) {
+				Write !, value_" is not a valid namespace to install IPM"_$Select(allowUpgrade: ". ", 1: ", or it already has IPM installed. ")_"Skipped."
+			} Else {
+				Set targetNamespaces(value) = 1
+			}
+		}
+	} Else {
+		// TODO: Implement this part
+		Write !, "Currently do not support interactively choose namespaces to install IPM. Use -namespaces or -globally modifier."
+		Return
+	}
+
+	// 4. Now that we got all the namespaces that need to install IPM, do the actual installation
+	If 'useLocal {
+		Set server = ##class(%IPM.Repo.Remote.Definition).DeploymentServerOpen(1,,.sc)
+		$$$ThrowOnError(sc)
+		If $IsObject(server) {
+			Set packageService = server.GetPackageService()
+			Set ipmRef = ##class(%IPM.Storage.ModuleInfo).%New()
+			// TODO: Make this ipm eventually once the package name is changed
+			Set ipmRef.Name = $$$IPMModuleName
+			If (targetVersion = "latest") {
+				// convert latest to semantic version so that it can be undestood by ModuleInfo class
+				Set server = ##class(%IPM.Repo.Remote.Definition).DeploymentServerOpen(1,,.sc)
+				$$$ThrowOnError(sc)
+				If $IsObject(server) {
+					Set targetVersion = server.GetPackageService().GetLatestIPMVersion()
+				}
+			}
+			Set ipmRef.VersionString = targetVersion
+			
+			// Retrieve module installer manifest
+			Set manifest = packageService.GetModuleInstallerManifest(ipmRef)
+			Set currentNS = ""
+			While 1 {
+				Set currentNS = $ORDER(targetNamespaces(currentNS))
+				If currentNS = "" {
+					Quit
+				}
+				New $Namespace
+				Set $Namespace = currentNS
+				Do $SYSTEM.OBJ.LoadStream(manifest, "ck/displayerror=0/displaylog=1")
+				Write !, "IPM enabled for namespace "_currentNS
+			}
+		} Else {
+			$$$ThrowOnError($$$ERROR($$$GeneralError,"Failed to get a valid registry server in order to download the IPM manifest."))
+		}
+	} Else {
+		Set targetIPMFileName = ipmLocalArray(targetVersion)
+		Set currentNS=$ORDER(targetNamespaces(""))
+		Set enabledNSList = ""
+		While (currentNS '="") {
+			Set $Namespace = currentNS
+			Set tSC = $SYSTEM.OBJ.Load(##class(%File).NormalizeFilename(targetIPMFileName, XMLDir), "ck")
+			If $$$ISOK(tSC) {
+				Set enabledNSList = enabledNSList_$LB(currentNS)
+			} else {
+				Write !, "IPM enabled for namespace(s) "_$ListToString(enabledNSList)
+				Write !, "IPM failed to enable for namespace "_currentNS
+				$$$ThrowOnError(tSC)
```

Comment:
Rather than throwing error, I suggest accumulating the errors and continuing so that IPM gets enabled in as many namespaces as possible. Then outside the loop, can indicate which namespaces it was successfully enabled/upgrade for and which ones it failed for. The same printing logic can be used for whether using remote or local artifact so can be outside the if-else block